express-notes-api/index.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const app = express();
+
+const fs = require('fs');
+const dataJson = require('./data.json');
+
+let nextID = 10;
+
+app.get('/api/notes', function (req, res) {
+  let key;
+  const arr = [];
+  for (key in dataJson.notes) {
+    arr.push(dataJson.notes[key]);
+  }
+  res.status(200).json(arr);
+
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  if (id < 0) {
+    res.status(400).send({ error: ' id must be a positive integer ' });
+  } else if (dataJson.notes[id] === undefined) {
+    res.status(404).send({ error: 'could not find the id' });
+  } else {
+    res.status(200).send(dataJson.notes[id]);
+  }
+
+});
+
+app.use(express.json());
+
+app.post('/api/notes', function (req, res) {
+
+  if (req.body.content !== undefined) {
+
+    req.body.id = nextID;
+    dataJson.notes[nextID] = req.body;
+    res.status(201).json(req.body);
```

Comment:
You should send this response in an `else` inside the `fs.writeFile` callback. The goal is to wait until the data is successfully written before confirming success with the client.